frontend/src/stores/roles/actions.ts

```diff
@@ -23,3 +23,23 @@ function setRoles(newRoles: Role[]): void {
 }
 
 export { fetchRoles, addRole }
+
+export const saveKnowledge = async (newKnowledge: string) => {
+  try {
+    const response = await fetch('/api/knowledges', {
```

Comment:
Don't use directly fetch. There is a method called `post` in http.ts file. Import it and use it

Reply:
done